core/include/gnuradio-4.0/PortTraits.hpp

```diff
@@ -50,6 +50,63 @@ using is_port = std::integral_constant<bool, is_port_v<Type>>;
 template<typename Collection>
 concept is_port_collection_v = is_port_v<typename Collection::value_type>;
 
+template<typename T>
+auto
+unwrap_port_helper() {
+    if constexpr (port::is_port_v<T>) {
+        return static_cast<T *>(nullptr);
+    } else if constexpr (port::is_port_collection_v<T>) {
+        return static_cast<typename T::value_type *>(nullptr);
+    } else {
+        meta::print_types<meta::message_type<"Is not a port or a collection of ports">, T>{};
+    }
+}
+
+template<typename T>
+using unwrap_port = std::remove_pointer_t<decltype(unwrap_port_helper<T>())>;
+
+struct port_kind {
```

Comment:
`gr::traits::port::port_kind::Stream` reads  bit redundant, maybe rename to `kind`?